### PR TITLE
DOC: Replace ggpy with plotnine in ecosystem

### LINKED ISSUE #31087: DOC: replace ggpy with plotnine on the ecosystem page
We have an entry about ggpy in the visualization section of the ecosystem page (https://dev.pandas.io/docs/ecosystem.html#visualization). However, ggpy is no longer maintained (last commit is more than 3 years ago: https://github.com/yhat/ggpy). While plotnine is an actively developed alternative ggplot-python-interface (https://github.com/has2k1/plotnine/). 

So I think we can just replace the entry about ggpy with one about plotnine.

### PR DESCRIPTION
**Pull request by hvardhan20**
- [ ] closes #31087 
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Changed ggpy reference to plotnine.

### CODE REVIEW

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -112,16 +112,16 @@ also goes beyond matplotlib and pandas with the option to perform statistical
 estimation while plotting, aggregating across observations and visualizing the
 fit of statistical models to emphasize patterns in a dataset.
 
-`yhat/ggpy <https://github.com/yhat/ggpy>`__
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+`has2k1/plotnine <https://github.com/has2k1/plotnine/>`__
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst]:
```suggestion
`plotnine <https://github.com/has2k1/plotnine/>`__
```

File: doc/source/ecosystem.rst [hunk 2]
```diff
@@ -112,16 +112,16 @@ also goes beyond matplotlib and pandas with the option to perform statistical
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Hadley Wickham's `ggplot2 <https://ggplot2.tidyverse.org/>`__ is a foundational exploratory visualization package for the R language.
 Based on `"The Grammar of Graphics" <https://www.cs.uic.edu/~wilkinson/TheGrammarOfGraphics/GOG.html>`__ it
 provides a powerful, declarative and extremely general way to generate bespoke plots of any kind of data.
-It's really quite incredible. Various implementations to other languages are available,
-but a faithful implementation for Python users has long been missing. Although still young
-(as of Jan-2014), the `yhat/ggpy <https://github.com/yhat/ggpy>`__ project has been
-progressing quickly in that direction.
+It's really quite incredible. Various implementations to other languages are available.
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst]:
```suggestion
Various implementations to other languages are available.
```

While editing this section, lets leave out that sentence (I don't think it adds value)

File: doc/source/ecosystem.rst [hunk 3]
```diff
@@ -112,16 +112,16 @@ also goes beyond matplotlib and pandas with the option to perform statistical
...
+A good implementation for Python users is `has2k1/plotnine <https://github.com/has2k1/plotnine/>`__,
+which started as an effort to improve the scaling functionality in
+`yhat/ggpy <https://github.com/yhat/ggpy>`__ formally known as "ggplot for python".
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst]: I am not fully sure if this historical note is needed.

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR!

**Comment by jreback** [comment]
thanks @hvardhan20

**Pull request merged**